### PR TITLE
Small cleanups in RecoTauTag/RecoTau

### PR DESCRIPTION
**Pull request by mbluj**
#### PR description:
This PR contains two small technical cleanups in the RecoTauTag/RecoTau package:
* removal from `python` directory of two cfi files which are generated automatically with the fillDescriptions mechanism,
* usage of a common helper function to get track of a tau charged hadron candidate instead of a duplication of the function functionality in one of plugins (code is using different logic, but results are identical).

This PR is technical and is expected to not change results.

#### PR validation:

Standard tests with `scram b runtests` and `runTheMatrix.py -l limited -i all --ibeos` successful (except a few tests which failed due to problem with accessing input files and which are considered as not relevant)

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by clacaputo** [comment]
@cmsbuild please test

File: RecoTauTag/RecoTau/plugins/PFRecoTauChargedHadronProducer.cc [hunk 1]
```diff
@@ -213,32 +213,7 @@ void PFRecoTauChargedHadronProducer::produce(edm::Event& evt, const edm::EventSe
...
-        }
-      }
-      if (track == nullptr) {
-        if (nextChargedHadron->getTrack().isNonnull()) {
-          track = nextChargedHadron->getTrack().get();
-        } else if (nextChargedHadron->getLostTrackCandidate().isNonnull()) {
-          track = nextChargedHadron->getLostTrackCandidate()->bestTrack();
-        }
-      }
+      const reco::Track* track = reco::tau::getTrackFromChargedHadron(*nextChargedHadron);
```
Comment (clacaputo) [inline RecoTauTag/RecoTau/plugins/PFRecoTauChargedHadronProducer.cc:216]: Hi @mbluj , do I understand right that the `getTrack()` used in `reco::tau::getTrackFromChargedHadron` here:

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/RecoTauTag/RecoTau/src/pfRecoTauChargedHadronAuxFunctions.cc#L14-L16

return the `_track` private member of  `reco::PFRecoTauChargedHadron`, filled here:

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/RecoTauTag/RecoTau/plugins/PFRecoTauChargedHadronFromPFCandidatePlugin.cc#L189-L201

and the logic is the same as the one previously used in the `PFRecoTauChargedHadronProducer`?

Comment (mbluj) [reply RecoTauTag/RecoTau/plugins/PFRecoTauChargedHadronProducer.cc:216]: The `getTrack()` function from `cmssw/RecoTauTag/RecoTau/src/pfRecoTauChargedHadronAuxFunctions.cc` returns either private member `track_` of `reco::PFRecoTauChargedHadron` or if the instance of  `reco::PFRecoTauChargedHadron` is built with `pat::PackedCandidate`s (on top of miniAOD) it returns `bestTrack()` of of the charged candidate (track built from embedded information).
Logic of the 'getTrack()` function is not identical to the logic of replaced code as the sequence of options which are checked is a bit different, but the final result is identical. Moreover, the `getTrack()` function is used also in other places of tau reconstruction code, so its use also in this place make the code more coherent.

Comment (clacaputo) [reply RecoTauTag/RecoTau/plugins/PFRecoTauChargedHadronProducer.cc:216]: Thanks for the explanation. Considering that the functionality was already implemented in `reco::tau::getTrackFromChargedHadron`, do you know why it was duplicated in `RecoTauTag/RecoTau/plugins/PFRecoTauChargedHadronProducer.cc`? Was there any specific reason?

Comment (mbluj) [reply RecoTauTag/RecoTau/plugins/PFRecoTauChargedHadronProducer.cc:216]: I think that there was not any specific reason, it was probably just forgotten when the code has been refactored in preparation for running with both reco::PFandidates (AOD mode) and pat::PackedCandidates (miniAOD mode).

**Comment by clacaputo** [comment]
+reconstruction

**Comment by perrotta** [comment]
+1

**Pull request merged**